base/linalg/triangular.jl

```diff
@@ -39,10 +39,15 @@ for t in (:LowerTriangular, :UnitLowerTriangular, :UpperTriangular, :UnitUpperTr
 
         real{T<:Real}(A::$t{T}) = A
         real{T<:Complex}(A::$t{T}) = (B = real(A.data); $t(B))
-
+        abs(A::$t) = $t(abs(A.data))
     end
 end
 
+imag(A::UpperTriangular) = UpperTriangular(imag(A.data))
+imag(A::LowerTriangular) = LowerTriangular(imag(A.data))
+imag(A::UnitLowerTriangular) = tril!(imag(A.data),-1))
```

Comment:
I think this will return a dense array if `A.data` is dense, so may still need the `LowerTriangular( )` wrapper around this